### DISCUSSION TITLE
Shovel connection fails and reports an authentication error

### DISCUSSION THREAD

**Discussion by NishantMunot28** [rabbitmq/rabbitmq-server#12108] [Questions] [answered]
### Describe the bug

[shovellog.txt](https://github.com/user-attachments/files/16727408/shovellog.txt)
1] shovel is working local to local on both the rabbitmq new and old.
2] shovel from old rabbit (**3.10.5**) to new (**3.12.13**) failed.
3] shovel from new rabbit (**3.12.13**) to old (**3.10.5**) failed
4] for testing I have created another rabbit with old version (**3.10.5**) and shovel is working for old version.

### Reproduction steps

1. Create a rabbitmq with version 3.10.5 add queue and vhost.
2. export definition from old rabbit 
3. create new rabbitmq with version 3.12.13.
4. import defination.
5. create shovel to migrate message from old rabbitmq queue to new rabbitmq queue

### Expected behavior

Shovel should migrate the messages to the new rabbitmq

**Comment by lukebakken** [comment]
https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support

Hello, thanks for using RabbitMQ.

RabbitMQ 3.10 and 3.12 are both out of support.

If you can demonstrate this issue while shoveling messages to RabbitMQ 3.13.6, please include _EVERYTHING_ we need to reproduce the issue, like exported definitions including the shovel itself. We won't guess how to reproduce this issue, even if it seems very simple.

I will re-open this discussion if you are able to reproduce with the latest version of RabbitMQ.

**Reply by NishantMunot28** [reply]
ok. Thanks for your quick response. one question, can I test shovel migration from 3.10.5 to 3.13.6 or it will still be unsupported version testing?

**Comment by johanrhodin** [comment, marked as answer]
`ACCESS_REFUSED - Login was refused using authentication mechanism PLAIN`
Check the username and password.

**Reply by lukebakken** [reply]
Thanks for looking at the log!

**Reply by michaelklishin** [reply]
Or whether `guest` is used to access a remote node, which is [prohibited by design](https://www.rabbitmq.com/docs/access-control#loopback-users) by default. [Dedicated users with generated credentials](https://www.rabbitmq.com/docs/production-checklist#users-and-permissions) should be used instead.

**Reply by NishantMunot28** [reply]
we have checked the username, password and access of the user on the vhost all seems correct, locally shovel is working with same username and passsword, getting error when using shovel from 3.10.5 to 3.12.13

**Reply by michaelklishin** [reply]
@NishantMunot28 the error message is very specific: your shovels fail to authenticate or have no permissions to access the target virtual host. 

A [very common reason for succeeding local and failing remote connections](https://www.rabbitmq.com/docs/access-control#loopback-users) was suggested.

There is a dedicated [doc section](https://www.rabbitmq.com/docs/access-control#troubleshooting-authn) for troubleshooting authentication.

We will not troubleshoot authentication for you. Please take it from here.

**Reply by michaelklishin** [reply]
Two other suggestions:

 * See [logs](https://www.rabbitmq.com/docs/logging) of the node (or all nodes in a cluster) that **accepts** shovel connections
 * `ACCESS_REFUSED` may mean **missing** permissions for accessing a virtual host and not incorrect credentials. See [Granting Permissions to a User](http://localhost:3000/docs/access-control#granting-permissions-to-a-user) and `rabbitmqctl help set_permissions`.

**Comment by michaelklishin** [comment]
@NishantMunot28 if you file one more duplicate issue you will be permanently banned from this org.

**Comment by lukebakken** [comment]
@NishantMunot28 please carefully read @michaelklishin's comment - https://github.com/rabbitmq/rabbitmq-server/discussions/12108#discussioncomment-10467165

I took about 30 minutes of my time to put together a _COMPLETE_ demonstration of setting up a shovel between a 3.10 and 3.12 node:

https://github.com/lukebakken/rabbitmq-server-12108?tab=readme-ov-file#demonstration

If you have questions about the above project, start a discussion here:

https://github.com/lukebakken/rabbitmq-server-12108/discussions

Do _NOT_ open any more issues concerning this topic in `rabbitmq/rabbitmq-server`, or you will be re-blocked.
